tests/python/gpu/test_kvstore_gpu.py

```diff
@@ -88,34 +89,48 @@ def check_rsp_pull(kv, count, ctxs, is_same_rowid=False, use_slice=False):
 
     # test fails intermittently. temporarily disabled till it gets fixed. tracked at https://github.com/apache/incubator-mxnet/issues/9384
     # check_rsp_push_pull('local')
+    os.environ["MXNET_KVSTORE_USETREE"] = ""
+    check_rsp_push_pull('device')
+    check_rsp_push_pull('device', is_push_cpu=False)
+    os.environ["MXNET_KVSTORE_USETREE"] = "1"
+    logging.info("Setting env to use tree reduce...")
     check_rsp_push_pull('device')
     check_rsp_push_pull('device', is_push_cpu=False)
 
 
 def test_row_sparse_pull_single_device():
-    kvstore = mx.kv.create('device')
-    copy = mx.nd.random_normal(shape=(4,4), ctx=mx.gpu(0))
-    grad = copy.tostype("row_sparse")
+    envs = ["","1"]
```

Comment:
minor suggestion: we could add some util class like https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/autograd.py#L93-L119 which manages the scope of such env var. It sets the env var to some var when entering the scope, and reset the env var when exiting the scope.